data-semantics.md

```diff
@@ -0,0 +1,117 @@
+
+It is not required that all suggested tags be used, even if one is used.
+
+### Component Identification
+
+For any span, it can be useful to specify the type of component being instrumented. This is particularly recommended for instrumentation provided for libraries or modules, where end-users may haev a mix of custom and library-provided instrumentation.
+
+* `component` - string
+    - Low-cardinality identifier of the module, library, or package that is instrumented.
+    - Examples:
+        - `httplib` for instrumentation of Python builtin httplib functionality
+        - `JDBC` for instrumentation of JDBC database connectors
+        - `mongoose` for instrumentation of Ruby MongoDB client connector
+
+
+### HTTP Server Tags
+
+These tags are recommended for spans marking entry into a HTTP-based service.
+
+* `http.url` - string
+    - URL of the request being handled in this segment of the trace, in [standard URI format](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier), excepting query string parameters.
+    - Protocol optional
+    - Examples:
+        - `https://domain.net/path/to?resource=here`
+        - `domain.net/path/to/resource`
+        - `http://user:pass@domain.org:8888`
+* `http.query` - string
+    - Query string parameters of the request being handled.
+    - Note that the preceding `?` character is not expected.
+    - Examples:
+        - `key=val&key2=val2`
+* `http.method` - string
+    - HTTP method of the request being handled.
+    - Case-insensitive
+    - Examples:
+        - `GET`, `POST`, `HEAD`
+* `http.status_code` - integer
+    - HTTP status code to be returned with HTTP response.
+    - Examples:
+        - `200`, `503`
+* `span.kind` - string
+    - Value of `s` should be used to indicate that this is a server side span (see "Peer Tags")
+
+
+### Peer Tags
+
+These tags can be provided by either client-side or server-side to describe the downstream (client case) or upstream (server case) peer being communicated with.
+
+* `peer.hostname` - string
+    - Remote hostname
+* `peer.ipv4` - string
+    - Remote IP v4 address
+* `peer.ipv6` - string
+    - Remote IP v6 address
+* `peer.port` - integer
+    - Remote port
+* `peer.service` - string
+    - Remote service name
+* `span.kind` - string
+    - One of `c` or `s`, indicating if this span represents a client or server
+
+
+
+## Semantic Log Use-Cases
+
+Some events are point-in-time and do not apply to an entire span; these are reported via logged events.  For instance, an error or exception may be reported via a `log` call.
+
+
+### Exception
+
+* Event: `exception`
+* Payload:
```

Comment:
this probably needs language examples - which structure is used to encode the payload?